index.js

```diff
@@ -0,0 +1,23 @@
+import express from 'express';
+import usersRouter from './routes/users.routes.js';
+import productsRouter from "./routes/products.routes.js";
+import cartsRouter from "./routes/carts.routes.js";
+
+const app = express();
+const port = process.env.PORT || 3001;
+
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
+app.use((req, _res, next) => (req.locals = {}) && next());
```

Comment:
why is this middleware needed?

Reply:
The main idea is to exclude code duplication and write code more clearly. It initializes an additional field that I use for storing data that will added/used in the next middlewares or actions(get, post, etc.). For example, I get the cart object in .findCart middleware and it becomes available in the cart to delete and put actions via req. locals.cart